### PR TITLE
Adds a skeletal implementaation of the framework up to the report gen…

### PR DESCRIPTION
**Pull request by dbernstein**
…eration.

The software will run against a Fedora3 repository.  It is not currently performing any validations.  Just looping through the FedoraObjectProcessors and  adding dummy ValidationResults.  No writing of results is happening yet.

### CODE REVIEW

**Review by awoods — CHANGES_REQUESTED** [review]
Can you segregate the interfaces and implementation classes into different packages?

File: src/main/java/org/fcrepo/migration/validator/AbstractValidationTaskBuilder.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package org.fcrepo.migration.validator;/*
```
Comment (awoods) [inline src/main/java/org/fcrepo/migration/validator/AbstractValidationTaskBuilder.java:1]: `package` should be after license statement.

File: src/main/java/org/fcrepo/migration/validator/F3ObjectValidationTask.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
+package org.fcrepo.migration.validator;
```
Comment (awoods) [inline src/main/java/org/fcrepo/migration/validator/F3ObjectValidationTask.java:1]: `package` should be after license statement.

File: src/main/java/org/fcrepo/migration/validator/F3ObjectValidationTaskBuilder.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package org.fcrepo.migration.validator;/*
```
Comment (awoods) [inline src/main/java/org/fcrepo/migration/validator/F3ObjectValidationTaskBuilder.java:1]: `package` should be after license statement.

File: src/main/java/org/fcrepo/migration/validator/F3SourceTypes.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package org.fcrepo.migration.validator;
```
Comment (awoods) [inline src/main/java/org/fcrepo/migration/validator/F3SourceTypes.java:1]: `package` should be after license statement.

File: src/main/java/org/fcrepo/migration/validator/Fedora3ObjectConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
+package org.fcrepo.migration.validator;/*
```
Comment (awoods) [inline src/main/java/org/fcrepo/migration/validator/Fedora3ObjectConfiguration.java:1]: `package` should be after license statement.

File: src/main/java/org/fcrepo/migration/validator/Fedora3ValidationConfig.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
+package org.fcrepo.migration.validator;/*
```
Comment (awoods) [inline src/main/java/org/fcrepo/migration/validator/Fedora3ValidationConfig.java:1]: `package` should be after license statement.

File: src/main/java/org/fcrepo/migration/validator/Fedora3ValidationExecutionManager.java [hunk 7]
```diff
@@ -0,0 +1,137 @@
+package org.fcrepo.migration.validator;/*
```
Comment (awoods) [inline src/main/java/org/fcrepo/migration/validator/Fedora3ValidationExecutionManager.java:1]: `package` should be after license statement.

File: src/main/java/org/fcrepo/migration/validator/FileSystemValidationResultWriter.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
+package org.fcrepo.migration.validator;/*
```
Comment (awoods) [inline src/main/java/org/fcrepo/migration/validator/FileSystemValidationResultWriter.java:1]: `package` should be after license statement.

**Review by awoods — CHANGES_REQUESTED** [review]

File: src/main/java/org/fcrepo/migration/validator/Driver.java [hunk 9]
```diff
@@ -37,33 +39,63 @@
...
+
+
     @CommandLine.Option(names = {"--debug"}, order = 30, description = "Enables debug logging")
     private boolean debug;
 
     @Override
     public Integer call() throws Exception {
-
-        return 0;
+        final var context = new AnnotationConfigApplicationContext("org.fcrepo.migration.validator");
```
Comment (awoods) [inline src/main/java/org/fcrepo/migration/validator/Driver.java:78]: Maybe open the `context` in the `try` clause... then remove the explicit `close` block below.

Comment (dbernstein) [reply src/main/java/org/fcrepo/migration/validator/Driver.java]: done.

File: src/main/java/org/fcrepo/migration/validator/impl/F3ObjectValidationTask.java [hunk 10]
```diff
@@ -0,0 +1,59 @@
+package org.fcrepo.migration.validator.impl;
```
Comment (awoods) [inline src/main/java/org/fcrepo/migration/validator/impl/F3ObjectValidationTask.java:1]: Please move `package` down.

**Review by awoods — CHANGES_REQUESTED** [review]

**Review by awoods — APPROVED** [review]

**Pull request merged**